packages/cypress/internal/cypress_repositories.bzl

```diff
@@ -0,0 +1,113 @@
+# Copyright 2020 The Bazel Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Repository rule used to install cypress binary.
+"""
+
+load("@bazel_tools//tools/build_defs/repo:http.bzl", "http_archive")
+
+def cypress_repositories(
+        name,
+        version,
+        linux_urls = [],
+        linux_sha256 = "",
+        darwin_urls = [],
+        darwin_sha256 = "",
+        windows_urls = [],
+        windows_sha256 = ""):
+    """
+    Repository rule used to install cypress binary.
+
+    Args:
+        name: Name of the external workspace where the cypress binary lives
+        version: Version of cypress binary to use. Should match package.json
+        linux_urls: (Optional) URLs at which the cypress binary for linux distros of linux can be downloaded. If omitted, https://cdn.cypress.io/desktop will be used.
+        linux_sha256: (Optional) SHA-256 of the linux cypress binary
+        darwin_urls: (Optional) URLs at which the cypress binary for darwin distros of linux can be downloaded. If omitted, https://cdn.cypress.io/desktop will be used.
+        darwin_sha256: (Optional) SHA-256 of the darwin cypress binary
```

Comment:
Does Cypress have more native binaries than just Linux, Darwin and Windows? eg arm vs amd? (esbuild does, so I'm not sure this pattern scales)

Reply:
They have the following binaries: 
"darwin-x64", "win32-ia32", "linux-x64", "win32-x64"

I opted not to include the win32-ia32 out of laziness